node/src/types/deploy.rs

```diff
@@ -1614,6 +1677,69 @@ pub struct DeployMetadata {
     pub execution_results: HashMap<BlockHash, ExecutionResult>,
 }
 
+// `bytesrepr::ToBytes` is not implemented for the inner `HashMap` because
+// bytesrepr is no-std and `HashMap` is only available in `std`.
+impl ToBytes for DeployMetadata {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        let mut buf = bytesrepr::allocate_buffer(self)?;
+        self.write_bytes(&mut buf)?;
+        Ok(buf)
+    }
+
+    fn serialized_length(&self) -> usize {
+        U32_SERIALIZED_LENGTH
+            + self
+                .execution_results
+                .iter()
+                .map(|(key, value)| key.serialized_length() + value.serialized_length())
+                .sum::<usize>()
+    }
+
+    fn write_bytes(&self, writer: &mut Vec<u8>) -> Result<(), bytesrepr::Error> {
+        let num_keys: u32 = self
+            .execution_results
+            .len()
+            .try_into()
+            .map_err(|_| BytesreprError::NotRepresentable)?;
+        num_keys.write_bytes(writer)?;
+        for (key, value) in self.execution_results.iter() {
```

Comment:
Can we sort the keys first? HashMap is unordered (and in stdlib hashmap specifically order is randomized) and two instances with the same data would generate different bytes. We should sort the keys first (or create a temporary `BTreeMap<&K, &V>` or smth) to make the representation deterministic.

Reply:
@mpapierski, are you concerned about data ending up in the trie at some point? In that case, having a different order to the bytes would affect hashing, as well as extra duplication of mutated data over time. Ideally this would be handled in the ToBytes impl for HashMap that @Fraser999 is suggesting.

Reply:
I'll wait for an answer here to push the change.

Reply:
@mpapierski done.